### ISSUE TITLE
Autocompleting end of environment like winedt

### ISSUE THREAD

**Issue by HelmerAslaksen** [texstudio-org/texstudio#3342]
### Describe the feature and the current behavior/state
I recently switched from Winedt to TeXStudio. With Winedt, if I had for instance

\begin{environment}
Some text
\end{

and I typed }, it would autocomplete to \end{environment}. I cannot find a way to do this with TeXstudio.

### Who will benefit with this feature?
I found this to be very useful, especially with nested environments.

**Comment by octaeder** [comment]
It's better to insert complete envs (begin+end). When you type say equ (without `\begin{`) and then press ctrl+alt+space the completer shows:

![image](https://github.com/texstudio-org/texstudio/assets/102688820/990c920a-bb97-42d3-90a1-7784de7440dc)

Press enter and the complete env is inserted. When you start typing \begin{equ after the second character (b) the completer opens automatically. Continue typing until one item is left. Then press enter.

See [manual](https://texstudio-org.github.io/editing.html#auto-completion).

**Comment by HelmerAslaksen** [comment]
Yes, I know, but often I am editing something I have already written, and then I decide to change which environment to use. And this may be nested inside another environment. In fact, it is also a very useful debugging tool. I think I'm inside a certain environment, but then it autocompletes to something else, and I realize there is a problem. Again, very useful when there are lots of nested environments. 

Just take my word for the fact that I've used this ability almost daily for many years with Winedt in addition to their version of the ability you describe. I'm forced to using TeXstudio because I'm using a machine owned by my university, and I'm missing this so much that trying to get the University to get a Winedt license.

**Comment by octaeder** [comment]
It's quite simple: Move the text cursor on a character of the env name, wait 1.5 seconds and the name gets framed. Now edit the name and your changes apply immediately to begin and end. There is no time to wait when you double click the name with the mouse. You may even use the completer to choose the new env name.

**Comment by octaeder** [comment]
s. [manual](https://texstudio-org.github.io/editing.html#special-commands) -> Rename environment

**Comment by HelmerAslaksen** [comment]
This is not what I'm looking for, but thanks for pointing it out, since there are situations where I could use this. If I have written
\begin{environment}
text
\end{e

then the begin part lights up. Is there any way to get the end part to complete? If I try ctrl-alt-space, it tries to insert a new begin.

**Comment by octaeder** [comment]
ctrl+alt+space for the completer, but you need to know what the env name is

**Comment by octaeder** [comment]
wait I thought it works but a second try does not...

**Comment by octaeder** [comment]
![image](https://github.com/texstudio-org/texstudio/assets/102688820/29fc546f-2943-41a2-bbb3-309b2e375598)

**Comment by octaeder** [comment]
Hm:

![image](https://github.com/texstudio-org/texstudio/assets/102688820/a78de7db-bce1-4aea-8ca7-da3065a06630)

In this case I need to delete backward until I end up with `\end`

**Comment by octaeder** [comment]
Can @sunderme clarify?

**Comment by sunderme** [comment]
not sure what exactly you are doing there. Works fine here.

additional method: alt+enter closes open env

**Comment by octaeder** [comment]
@HelmerAslaksen Instead of completing `\end{eq` (where the cursor is at the end of the text) with ctrl+alt+space just use 'ctrl+space'. This will open the completer with `\end{equation}`.

@sunderme Pressing alt+enter adds `\end{equation}` independent of the fact that `\end{eq` is present before the cursor.

@sunderme After completing an env as usual (steps 1 and 2),  delete the placeholder with del and move the cursor to the end of the env (step 3).  Use backspace to delete most of the envname parameter of \end (step 4). Now press ctrl+alt+space (step 5).

![image](https://github.com/texstudio-org/texstudio/assets/102688820/c4f80e5c-8230-4e0f-b06d-8fbd89a944f5)

As said above using ctrl+space does the right job.

**Comment by sunderme** [comment]
this should be answered

**Comment by HelmerAslaksen** [comment]
Thanks @octaeder! Ctrl+Space works for me!
